workbench-core/accounts/src/utilities/projectUtils.test.ts

```diff
@@ -0,0 +1,526 @@
+
+describe('projectUtils', () => {
+  let projects: Project[];
+
+  beforeEach(() => {
+    projects = [
+      {
+        id: '',
+        name: 'name1',
+        description: '',
+        costCenterId: 'cc-1',
+        status: ProjectStatus.AVAILABLE,
+        createdAt: '2022-11-10T04:19:00.000Z',
+        updatedAt: '',
+        awsAccountId: '',
+        envMgmtRoleArn: '',
+        hostingAccountHandlerRoleArn: '',
+        vpcId: '',
+        subnetId: '',
+        environmentInstanceFiles: '',
+        encryptionKeyArn: '',
+        externalId: '',
+        accountId: ''
+      },
+      {
+        id: '',
+        name: 'name2',
+        description: '',
+        costCenterId: 'cc-2',
+        status: ProjectStatus.SUSPENDED,
+        createdAt: '2022-11-10T04:20:00.000Z',
+        updatedAt: '',
+        awsAccountId: '',
+        envMgmtRoleArn: '',
+        hostingAccountHandlerRoleArn: '',
+        vpcId: '',
+        subnetId: '',
+        environmentInstanceFiles: '',
+        encryptionKeyArn: '',
+        externalId: '',
+        accountId: ''
+      },
+      {
+        id: '',
+        name: 'name3',
+        description: '',
+        costCenterId: 'cc-3',
+        status: ProjectStatus.DELETED,
+        createdAt: '2022-11-10T04:21:00.000Z',
+        updatedAt: '',
+        awsAccountId: '',
+        envMgmtRoleArn: '',
+        hostingAccountHandlerRoleArn: '',
+        vpcId: '',
+        subnetId: '',
+        environmentInstanceFiles: '',
+        encryptionKeyArn: '',
+        externalId: '',
+        accountId: ''
+      }
+    ];
+  });
+
+  describe('manualSortProjects', () => {
+    let sortRequest: ProjectSort;
+    let expectedProjects: Project[];
+
+    test('should return original list if no sort passed in', async () => {
+      expect(manualSortProjects(sortRequest, projects)).toEqual(projects);
+    });
+
+    test('should fail if too many sort attributes passed', async () => {
+      // BUILD
+      sortRequest = { createdAt: 'asc', name: 'asc' };
+
+      // TEST n CHECK
+      expect(() => manualSortProjects(sortRequest, projects)).toThrow(
+        'Cannot sort by more than one attribute.'
+      );
+    });
+
+    test('should fail if sortValue is undefined', async () => {
+      // BUILD
+      sortRequest = { createdAt: undefined };
+
+      // TEST n CHECK
+      expect(() => manualSortProjects(sortRequest, projects)).toThrow('Sort contains invalid format');
+    });
+
+    test('should fail if sortKey is not property on project', async () => {
+      // BUILD
+      sortRequest = { notAnAttribute: 'asc' };
+
+      // TEST n CHECK
+      expect(() => manualSortProjects(sortRequest, projects)).toThrow(
+        `No notAnAttribute on Project. Please sort by another attribute.`
+      );
+    });
+
+    test('sort on createdAt ascending', async () => {
+      // BUILD
+      sortRequest = { createdAt: 'asc' };
+      expectedProjects = projects;
```

Comment:
I am not sure what you mean by this comment. Can you elaborate?

Reply:
`projects` is only in the order that its in because of the initial definition. If someone were to change that definition, the sort order would be wrong and the test would fail. By sorting your data as a part of the setup for the test, you avoid depending on the initial definition being in the right order.

Reply:
We will leave as is but if someone needs to add new entities to the list of projects, they need to verify it doesn't break tests when they do so. I don't really expect someone to add a new project in the set up though.